edgedb/datatypes/tuple.c

```diff
@@ -172,6 +172,98 @@ tuple_getitem(EdgeTupleObject *o, Py_ssize_t i)
     return el;
 }
 
+static PyObject *
+tuple_getslice(EdgeTupleObject *o, Py_ssize_t start, Py_ssize_t stop, Py_ssize_t step) {
+    Py_ssize_t size = Py_SIZE(o);
+
+    // negative indexes count from the end
+    if (start < 0)
+        start += size;
+    if (stop < 0)
+        stop += size;
+
+    if (step < 0) {
+        if (stop < -1)
+            stop = -1;
+        if (start > size - 1)
+            start = size - 1;
+
+        if (start <= stop)
+            return EdgeTuple_New(0);
+    } else {
+        if (start < 0)
+            start = 0;
+        if (stop > size)
+            stop = size;
+
+        if (stop <= start)
+            return EdgeTuple_New(0);
+
+        if (start == 0 && stop == size && step == 1) {
+            Py_INCREF(o);
+            return o;
+        }
+    }
+
+    Py_ssize_t new_size = 0;
+    if (step < 0) {
+        for (Py_ssize_t i = start; i > stop; i += step) {
```

Comment:
I'm pretty sure there is a better way than this to calculate the new tuple's size...

Reply:
See how Python does this: https://github.com/python/cpython/blob/0a40025b803a8dc1d604538e317992827ab96625/Objects/tupleobject.c#L779